### PR TITLE
[CPU][POC] Mixed Affinity prototype

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
This PR introduces transformation passes that allow to decompose batched part of a model into several parallel branches (based on heuristic algorithms) with lower batch. Introduced passes:
 - *MarkupOptimalBS: pass which marks layers with optimal batch size (using rt_info) based on heuristics. Heuristics is in progress yet and might be changed*
 - *PropagateOptimalBS: propagate optimal batch size through the graph to form subgraphs with equal optimal batch size. Also calculates the number of clones that must be created for each layer*
 - *formSubgraphs: auxiliary function which analyses marked model and collect inputs and outputs for each subgraph*
 - *SwitchAffinity: pass which using `formSubgraphs` results decomposes each subgraph into several parallel branches with lower batch*

### Tickets:
 - *77278*
 - *102528*

### CODE REVIEW

**Comment by v-Golubev** [comment]
@jane-intel could you please take a look on transformations part? Thanks!

File: src/plugins/intel_cpu/src/ngraph_transformations/propagate_optimal_bs.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        // batch_label propagation
+        node->validate_and_infer_types();
+        if (std::all_of(out_shape.begin(), out_shape.end(),
+            [](const ov::Dimension& d) { return ov::DimensionTracker::get_label(d) != batch_label; })) {
+            continue;
+        }
+
+        const size_t out_batch_idx = get_batch_idx(out_shape);
+        const auto out_batch = out_shape[out_batch_idx].get_length();
+        for (const auto& input : node->input_values()) {
```
Comment (jane-intel) [inline src/plugins/intel_cpu/src/ngraph_transformations/propagate_optimal_bs.cpp:57]: what happens with non constant inputs that don't deliver batch label? general question -- no concern yet

**Review by jane-intel — COMMENTED** [review]
Reviewed transformation part and label usage. From this standpoint everything is used as intended.
Additionally, I expressed several concerns via IMs to Vladislav. He will be testing those cases and see if changes are needed.
Looking forward to see more strict checks if shape is fully static and general refactoring of repeated code.
P.S. Commenting your code won't hurt

**Comment by akladiev** [comment]
This PR was closed because it has been stalled for 2 week with no activity.

**Pull request closed without merging**